tools/wpt/browser.py

```diff
@@ -939,44 +944,214 @@ class Chrome(ChromeChromiumBase):
+        if len(downloads_for_platform) == 0:
+            self.logger.info(f"No ChromeDriver download found for build {version}"
+                            f"of platform {self.platform}")
+            return None
+        return downloads_for_platform[0]["url"]
 
-    def download(self, dest=None, channel=None, rename=None):
-        raise NotImplementedError("Downloading of Chrome browser binary not implemented.")
+    def _get_download_urls_by_version(self, version):
+        """Find Chrome for Testing and ChromeDriver download URLs matching a specific version.
+
+        Returns: Both binary downloads for Chrome and ChromeDriver.
+        """
+        try:
+            # Get a list of versions with download URLs from Chrome for Testing.
+            resp = get(
+                "https://googlechromelabs.github.io"
+                "/chrome-for-testing/known-good-versions-with-downloads.json")
+        except requests.RequestException as e:
+            raise requests.RequestException(
+                "Chrome for Testing versions not found", e)
+        versions_json = resp.json()
+        versions_list = versions_json["versions"]
+        # Attempt to find a version match in the list of available downloads.
+        matching_versions = [v for v in versions_list if v["version"] == version]
+        if len(matching_versions) == 0:
+            raise DownloadNotFoundError(f"No Chrome for Testing download found for {version}")
+
+        download_info = matching_versions[0]["downloads"]
+        # Find the download url that matches the current platform.
+        browser_download_urls = [d for d in download_info["chrome"]
+                                 if d["platform"] == self._chrome_platform_string]
+        if len(browser_download_urls) == 0:
+            raise DownloadNotFoundError(
+                f"No Chrome for Testing download found for {self.platform} of version {version}")
+        browser_download_url = browser_download_urls[0]["url"]
+
+        # Get the corresponding ChromeDriver download URL for later use.
+        chromedriver_download_urls = [d for d in download_info["chromedriver"]
+                                      if d["platform"] == self._chrome_platform_string]
+        if len(chromedriver_download_urls) == 0:
+            # Some older versions of Chrome for Testing
+            # do not have a matching ChromeDriver download.
+            raise DownloadNotFoundError(
+                f"ChromeDriver download does not exist for version {version}")
+        chromedriver_url = chromedriver_download_urls[0]["url"]
+
+        return browser_download_url, chromedriver_url
+
+    def _get_download_urls_by_channel(self, channel):
+        """Find Chrome for Testing and ChromeDriver download URLs matching the given channel.
+
+        Returns: Both binary downloads for Chrome and ChromeDriver.
+        """
+        try:
+            resp = get(
+                "https://googlechromelabs.github.io"
```

Comment:
This can be moved to a constant at the top of the file.

Reply:
Agreed. I've moved the root URL to the top and use it for any CfT requests. 🙂